java/client/src/main/java/com/youtube/vitess/client/StreamIterator.java

```diff
@@ -0,0 +1,37 @@
+package com.youtube.vitess.client;
+
+import java.util.NoSuchElementException;
+
+/**
+ * StreamIterator is used to access the results of a streaming call.
+ *
+ * It is similar to java.util.Iterator, but the hasNext() method is
+ * understood to block until either a result is ready, an error occurs,
+ * or there are no more results. Also, unlike Iterator, these methods
+ * can throw VitessException or VitessRpcException.
+ */
+public interface StreamIterator<E> {
```

Comment:
The reason I couldn't use the built-in Iterator interface is because ours needs to throw checked exceptions to make the user handle errors. Maybe an iterator-like interface is the wrong way to go here, but the suggestion from @sougou was a "fetch next" style API, so I figured I'd make it look similar to Iterator for familiarity.